test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/internal/CreateMockitoSpiesCallback.java

```diff
@@ -1,20 +1,33 @@
 package io.quarkus.test.junit.mockito.internal;
 
 import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.Set;
 
 import org.mockito.AdditionalAnswers;
 import org.mockito.Mockito;
 
+import io.quarkus.arc.Arc;
 import io.quarkus.arc.ClientProxy;
+import io.quarkus.arc.InjectableContext;
 import io.quarkus.arc.InstanceHandle;
+import io.quarkus.test.junit.callback.QuarkusTestAfterAllCallback;
 import io.quarkus.test.junit.callback.QuarkusTestAfterConstructCallback;
+import io.quarkus.test.junit.callback.QuarkusTestContext;
 import io.quarkus.test.junit.mockito.InjectSpy;
 
-public class CreateMockitoSpiesCallback implements QuarkusTestAfterConstructCallback {
+public class CreateMockitoSpiesCallback implements QuarkusTestAfterConstructCallback, QuarkusTestAfterAllCallback {
+
+    // in nested tests, there are multiple states created before destruction is triggered
+    private static Set<InjectableContext.ContextState> statesToDestroy = new HashSet<>();
 
     @Override
     public void afterConstruct(Object testInstance) {
         Class<?> current = testInstance.getClass();
+        boolean contextPreviouslyActive = Arc.container().requestContext().isActive();
```

Comment:
As `QuarkusTestAfterConstructCallback` can be used in `@QuarkusIntegrationTest` as well where no `Arc` exists, we are going to need to be defensive against `Arc.container()` being null

Reply:
Ah, good point, thanks.
Should be fixed now and added a comment mentioning the reason.